### ISSUE TITLE
ORKNumericAnswerFormat is processor-platform dependent for Integers

### ISSUE THREAD

**Issue by md0u80c9** [ResearchKit/ResearchKit#534]
Attached is a definition for an NHS Number (UK NHS Numbers are 10 digits long; although they are a bit smarter than modelled below, the bottom line is that they are always 10 digits long and always start with a 4 or a 5).

The following definition for ORKNumericAnswerFormat works when placed into an iPhone 6/6S, but the application crashes when tested on an iPad 2. I'm presuming the reason is that the former are 64-bit, and the latter are 32-bit.

let nhsNoFormat = ORKNumericAnswerFormat(style: ORKNumericAnswerStyle.Integer, unit: nil, minimum: 4000000000, maximum: 9999999999)

**Comment by YuanZhu-apple** [comment]
I am not sure how `swift` works with auto boxing `NSNumber`, can you try to init the NSNumber with `initWithLongLong:` ?

**Comment by umerkhan-apple** [comment]
@md0u80c9 Are you still experiencing this issue?

**Comment by md0u80c9** [comment]
Yeah it's still a bit of a challenge: I ended up removing the numerical limiter on the answer format which is the wrong thing to do. I didn't quite get NSNumber to do what I wanted for some reason (can't remember how/why this didn't work now).

Is it possible to define ORKNumericAnswerFormat as always being a 64-bit integer? On 64-bit platforms this will be the compilation default anyway, but it might help with devices like the iPhone 5C or 1st gen iPad Mini.

**Comment by Pariecemckinney-apple** [comment]
@md0u80c9 Just wanted to check in to see if there was any progress made on this issue. I would also presume that the 32bit devices are the root cause of this issue. Not sure how this could be handled with the max value for even UInt32 being only 4294967295 (much lower than what you need).

**Comment by DanKeen** [comment]
Personally, I'd treat the NHS number as a string. It's a 3-3-4, right? And there's a possibility of leading zeros - nothing in the standard precludes that (afaict).

**Comment by DanKeen** [comment]
(Not that that resolves the underlying issue, of course…)

**Comment by md0u80c9** [comment]
Hi Dan. I currently use it as a string. However, that means you are presented with a standard keyboard (rather than the numeric one), which of course is inconvenient to the user given all the values are digits.

The easy solution is to drop 32-bit support - which will happen naturally soon anyway. However there are a few iPad Mini 1st gens knocking around our department...

There aren't leading digits on the NHS number (they haven't allocated a chunk starting with 0): and there is a calculation which permits you to do a checksum on it (can't remember the link). So having a 64-bit Int may be easier for dissecting the value to check the checksum.

**Comment by DanKeen** [comment]
Ugh, didn't consider that switching to a string means you don't get a nice keyboard. Boo :( Perhaps that's worth a feature request as well?

I know NHS has only assigned starting with 4 and 5 (as you reference above), but there is a huge block reserved for testing starting at 010, right?

I think ideally ORKNumericAnswerFormat should be platform-agnostic and backed by some kind of bignum representation. But that's probably something that we'd want to do more widely than just ORKNumericAnswerFormat, and would be an API-breaking change even if we just changed it here.
